varcode/common.py

```diff
@@ -59,3 +59,14 @@ def trim_shared_flanking_strings(ref, alt):
     ref, alt, prefix = trim_shared_prefix(ref, alt)
     ref, alt, suffix = trim_shared_suffix(ref, alt)
     return ref, alt, prefix, suffix
+
+
+def group_by(records, field_name):
```

Comment:
1) Yep, I didn't want the small annoyance of converting the iterator to a dictionary at the call site. 

2) Good point about `defaultdict`!